cmd/appgw-ingress/main.go

```diff
@@ -93,19 +93,26 @@ func main() {
 	azContext, err := azure.NewAzContext(env.AzContextLocation)
 	if err != nil {
 		glog.Info("Unable to load Azure Context file:", env.AzContextLocation)
+	} else if azContext.VNetResourceGroup == "" {
+		azContext.VNetResourceGroup = azContext.ResourceGroup
 	}
 
 	// adjust env variable
-	if env.AppGwName == "" {
-		env.AppGwName = env.ReleaseName
+	if env.AppGwResourceID != "" {
+		subscriptionID, resourceGroupName, applicationGatewayName := azure.ParseResourceID(env.AppGwResourceID)
+		env.SubscriptionID = string(subscriptionID)
+		env.ResourceGroupName = string(resourceGroupName)
+		env.AppGwName = string(applicationGatewayName)
```

Comment:
We could also change the type of these env struct members from `string` to `SubscriptionID` etc.